compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/util/Unifier.java

```diff
@@ -1161,6 +1167,15 @@ private BType getExpectedTypeForInferredTypedescMember(BUnionType originalType,
             return null;
         }
 
+        // Add the original union type if all the members of the original type are present in the `types` list.
+        for (BType originalMemberType : expUnionType.getOriginalMemberTypes()) {
+            LinkedHashSet flatTypeSet = toFlatTypeSet(new LinkedHashSet<>(Set.of(originalMemberType)));
+            if (flatTypeSet.stream().filter(type -> types.contains(type)).count() == flatTypeSet.size()) {
```

Comment:
Can't this be simplified to a `containsAll`?

Reply:
We can thanx